modules/ding_library/ding_library.module

```diff
@@ -216,7 +216,13 @@ function ding_library_disable_custom_alias($form, &$form_state) {
     $form['path']['pathauto']['#description'] = t('Custom alias has been disabled');
     $form['path']['pathauto']['#attached']['js'][] = drupal_get_path('module', 'ding_library') . '/ding_library.js';
 
-    unset($form['path']['alias']);
+    // We have to keep the alias textfield, since removing it causes a fatal
+    // error in the parent element validate handler, so we just tell the FORM
+    // API it's disabled and make sure it's disabled in frontend.
+    $form['path']['alias']['#disabled'] = TRUE;
+    // For some reason this is necessary.
+    $form['path']['alias']['#attributes'] = array('disabled' => TRUE);
+    unset($form['path']['alias']['#states']);
```

Comment:
I am not very happy about the existing solution with JavaScript used to faking a checkbox. This change only seems to pile on top of the problem making a bad situation worse.

I see from previous changes that setting `#access` to `FALSE` has been tried - also by yourself in PR #730. To summarize the problem with that solution is that if new library nodes are created then they will not have a path alias generated by `pathauto` either.

I tried investigation the solution and found [this issue](https://www.drupal.org/project/pathauto/issues/2110365#comment-9250617) which seems to describe our problem exactly and provide a solution, which would allow us to remove all other workarounds.

Could you test this out?

Reply:
If we have to keep this it is worth noting:

1. `#states` is set by `pathauto`
2. Adding this disabled attribute manually might be necessary because Drupal Core would normal do this by checking for `#disabled` elements before be set this ourselves.
3. This overwrites `#attributes` completely. It should be sufficient to merge in `[ 'disabled' => TRUE]`

Reply:
@kasperg Much better solution :)

I just tested it and it works, so I updated the PR with new solution.

Reply:
@holt83 Thanks. Appreciate the effort.